### ISSUE TITLE
App.py

### ISSUE THREAD

**Issue by Mreed87** [Mreed87/Streamlit-app_py#2]
import streamlit as st
import yfinance as yf
from prophet import Prophet
import pandas as pd
import plotly.express as px

st.set_page_config(page_title="Penny Stock Forecaster", layout="wide")

st.title("📈 Hourly Penny Stock Forecaster")
st.write("Forecast short-term movements of penny stocks using hourly data + Prophet.")

# User input
ticker = st.text_input("Enter Penny Stock Ticker (example: SNDL)", "SNDL")
horizon = st.slider("Forecast Hours Ahead", 1, 48, 12)

if st.button("Run Forecast"):
    st.write(f"📡 Fetching latest hourly data for **{ticker}**...")

    # Load data
    try:
        data = yf.download(ticker, period="5d", interval="1h")
    except Exception as e:
        st.error(f"Data download failed: {e}")
        st.stop()

    if data.empty:
        st.error("⚠️ No data found! Try another ticker symbol.")
    else:
        df = data.reset_index()[["Datetime", "Close"]]
        df.rename(columns={"Datetime": "ds", "Close": "y"}, inplace=True)

        # Train Prophet model
        model = Prophet()
        model.fit(df)

        # Forecast future
        future = model.make_future_dataframe(periods=horizon, freq="H")
        forecast = model.predict(future)

        # Chart forecast
        fig = px.line(forecast, x="ds", y="yhat",
                      title=f"{ticker} Forecast (Next {horizon} Hours)",
                      labels={"ds": "Date/Time", "yhat": "Predicted Price"})
        fig.add_scatter(x=df["ds"], y=df["y"], mode="markers", name="Actual Prices")

        st.plotly_chart(fig, use_container_width=True)

        # Show forecast table
        st.subheader("🔮 Forecast Data")
        st.dataframe(forecast[["ds", "yhat", "yhat_lower", "yhat_upper"]].tail(horizon))
